iceberg/azure/src/main/java/org/apache/iceberg/azure/adlsv2/ADLSFileIO.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.azure.adlsv2;
```

Comment:
Ditto. @jiang95-dev / @abhisheknath2011  can you help with the packaging?

Reply:
I think we can have package as `com.linkedin.openhouse.azure.adlsv2`

Reply:
Nvm. I got the context from Kai why we are copying code. Either way is ok.

Reply:
I asked Kai to send explore git submodules instead of copying code.

Reply:
Sure, sounds good. Yes copying is required due to iceberg version compatibility issue.

Reply:
hmm..I am interested in this as well.

Reply:
For an update, it seems like you can only submodule an entire repository, not a subfolder. Is that still worth it? There were some sketchy solutions that I tried to get just a subfolder, but none of them worked for me.